marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/impl/hub-entities/generateIndexes.sjs

```diff
@@ -232,22 +232,89 @@ function generateIndexConfigWithStructuredProperties() {
       "definitions": {
         "Book": {
           "properties": {
-            "title": {"datatype": "string", "facetable": true, "collation": "http://marklogic.com/collation/"},
-            "authors": {"datatype": "array", "facetable": true, "items": {"datatype": "string"}},
-            "rating": {"datatype": "integer", "facetable": true, "items": {"datatype": "string"}},
-            "id": {"datatype": "string", "facetable": false, "collation": "http://marklogic.com/collation/"},
-            "customer": {"datatype": "array", "items": {"$ref": "#/definitions/Address"}, "facetable": true},
-            "address": {"$ref": "#/definitions/Address", "facetable": true}
+            "title": {
+              "datatype": "string",
+              "facetable": true,
+              "collation": "http://marklogic.com/collation/"
+            },
+            "authors": {
+              "datatype": "array",
+              "facetable": true,
+              "items": {
+                "datatype": "string"
+              }
+            },
+            "rating": {
+              "datatype": "integer",
+              "facetable": true,
+              "items": {
+                "datatype": "string"
+              }
+            },
+            "id": {
+              "datatype": "string",
+              "facetable": false,
+              "collation": "http://marklogic.com/collation/"
+            },
+            "customer": {
+              "datatype": "array",
+              "items": {
+                "$ref": "#/definitions/Address"
+              }
+            },
+            "address": {
+              "$ref": "#/definitions/Address"
+            }
+          }
+        },
+        "Address": {
+          "properties": {
+            "street": {
```

Comment:
I would simplify this model by removing everything you're not testing against, as it's just noise - so remove "street". But "state" is useful in that we want to verify that there's not a range index generated for it.